ciao-sdk/show.go

```diff
@@ -0,0 +1,459 @@
+package sdk
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/ciao-project/ciao/ciao-controller/api"
+	"github.com/ciao-project/ciao/ciao-controller/types"
+	"github.com/ciao-project/ciao/openstack/compute"
+	"github.com/ciao-project/ciao/openstack/image"
+	"github.com/ciao-project/ciao/payloads"
+
+	"github.com/intel/tfortools"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"net/http"
+)
+
+type byCreated []compute.ServerDetails
+
+func (ss byCreated) Len() int           { return len(ss) }
+func (ss byCreated) Swap(i, j int)      { ss[i], ss[j] = ss[j], ss[i] }
+func (ss byCreated) Less(i, j int) bool { return ss[i].Created.Before(ss[j].Created) }
+
+func listEvent(cmd *cobra.Command, args []string) error {
```

Comment:
I know we've agreed to create generic commands like show and delete that can apply to a range of objects.  However, from a code organisation point of view I think it would make more sense to keep the code grouped by object type rather than by action.  So node.go would have all the node functionality, instance.go the instance functionality, just as before.  show.go would then just contain the Show function that calls into all the other modules.

Reply:
Also, I think the sdk package is currently mixing up two things.

1. It contains the code to formulate requests and parse responses for REST APIs
2. It contains cli code for building a cli.

I think these two things need to be separate.  If I was going to create my own tool to issue commands to the ciao cluster I probably wouldn't want any of the CLI stuff.  So I think this pacakge needs to be split into two.  Keep the REST stuff in here and move the cli stuff into a package under ciao/ciao

Reply:
Actually, it already looks like you have a package for 2, ciao/cmd.